traptor/traptor.py

```diff
@@ -593,11 +608,12 @@ def _main_loop(self):
 
                     if self.kafka_enabled == 'true':
                         try:
-                            self.kafka_conn.send(self.kafka_topic, enriched_data)
-                            self.logger.info("Tweet sent to kafka: {}".format(tweet.get('id_str', None)))
-                            dd_monitoring.increment('tweet_to_kafka_success')
+                            self.logger.info("Attempting to send tweet {} to kafka".format(tweet.get('id_str', None)))
+                            future = self.kafka_conn.send(self.kafka_topic, enriched_data)
+                            future.add_callback(self.kafka_success_callback, tweet)
+                            future.add_errback(self.kafka_failure_callback)
                         except Exception:
-                            self.logger.error("Unable to add tweet to Kafka: {}".format(traceback.format_exc()))
+                            self.logger.error("Uncaught exception sending tweet to Kafka: {}".format(traceback.format_exc()))
                             dd_monitoring.increment('tweet_to_kafka_failure')
```

Comment:
@joequery with the `def _gen_kafka_failure` message performing an increment of the `tweet_to_kafka_failure` in DD, is it needed in the Exception clause?